src/ui_fsmenu/menu.h

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (C) 2002-2020 by the Widelands Development Team
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ */
+
+#ifndef WL_UI_FSMENU_MENU_H
+#define WL_UI_FSMENU_MENU_H
+
+#include "ui_basic/box.h"
+#include "ui_basic/button.h"
+#include "ui_basic/window.h"
+#include "ui_fsmenu/main.h"
+
+namespace FsMenu {
+constexpr int kPadding = 4;
+
+class BaseMenu : public UI::Window {
```

Comment:
Please write a few lines of documentation for each class in this file: What does the respective menu look like? What widgets and features does it offer or lack? How to choose which one to use when implementing a new window?